src/trio/_core/_run.py

```diff
@@ -2075,6 +2133,7 @@ async def init(
                     )
 
                 # Main task is done; start shutting down system tasks
+                # TODO: source/reason?
```

Comment:
Hmm, maybe we should have a `trio_shutdown` source that we can use here and for asyncgen shutdown?

Reply:
I'm not sure about the `trio_` prefix - if it's to indicate "trio itself is shutting down" then it should not be used for asyncgen shutdown; if it's to indicate "the source is from within trio" then most of the other sources should also have it, and only really makes sense within a context of the user adding a bunch of their own sources...

which we *could* allow, you can currently do
```python
my_cs._cancel_reason = CancelReason(source="foo", reason="bar", source_task="baz")
my_cs.cancel()
```
we could also open up more parameters to `.cancel()`.

but giving an extensive message to `reason` should be enough for most cases tbh.

Reply:
Agree, let's just remove this TODO comment, and the one below.